app/src/main/java/com/example/criminalintentapp/presentation/LoginActivity.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.criminalintentapp.presentation
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import com.example.criminalintentapp.databinding.ActivityLoginBinding
+import com.google.firebase.auth.FirebaseAuth
+
+class LoginActivity : AppCompatActivity() {
+
+    private val authenticationViewModel: AuthenticationViewModel by lazy {
+        ViewModelProvider(this).get(AuthenticationViewModel::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val user = FirebaseAuth.getInstance().currentUser
+        if (user != null) {
+            val intent = Intent(this@LoginActivity, MainActivity::class.java)
+            startActivity(intent)
+            finish()
+        }
+
+        authenticationViewModel.userLoginLiveData.observe(this) { user ->
+            if (user != null) {
+                Toast.makeText(this, "Logged in successfully", Toast.LENGTH_SHORT).show()
+
+                val intent = Intent(this@LoginActivity, MainActivity::class.java)
```

Comment:
You could add this code inside a `goToMainActivity(clearBackStack: Boolean)` function. This would be called twice, once with `clearBackStack = true` where flags are added to the intent and once with `clearBackStack = false` -> meaning no flags are added